src/main/scala/com/exasol/cloudetl/kafka/KafkaTopicDataImporter.scala

```diff
@@ -67,9 +68,17 @@ object KafkaTopicDataImporter extends LazyLogging {
             record.partition().asInstanceOf[AnyRef],
             record.offset().asInstanceOf[AnyRef]
           )
-          val avroRow = AvroRow(record.value()).getValues().map(_.asInstanceOf[AnyRef])
-          val exasolRow: Seq[Object] = avroRow ++ metadata
-          iterator.emit(exasolRow: _*)
+
+          if (singleColJson) {
+            val rec = record.value()
+            val exasolRow: Seq[Object] = Seq(s"$rec") ++ metadata
+            iterator.emit(exasolRow: _*)
+          } else {
+            val avroRow = AvroRow(record.value()).getValues().map(_.asInstanceOf[AnyRef])
+            val exasolRow: Seq[Object] = avroRow ++ metadata
+            iterator.emit(exasolRow: _*)
+          }
```

Comment:
Could you please add one private function for a row?
Suggestion:
```scala
val exasolRow: Seq[Object] = getAvroRow(singleColumnJson) ++ metadata
iterator.emit(exasolRow: _*)
```
where the `getAvroRow` function contains the if else part.